src/main/java/com/example/petbuddybackend/service/chat/session/ChatSessionService.java

```diff
@@ -2,101 +2,190 @@
 
```

Comment:
Nwm czy niebylbym za tym, zeby wydzielic osobna klase do samego wysylania wiadomosci i zajmowania sie ogolnie wysylką, a zarzadzanie sesjami, subskrybcjami dac osobno. Wydaje mi sie, ze byloby to bardziej przejrzyste. Więc w tym ChatSessionService zmienilbym nazwe na np. ChatMessageSenderService, a wtedy ChatMessageSenderService nie mialby ChatSessionTracker, zamiast tego ChatSessionTracker mialby controller websocketowy i w kontrollerze bezposrednio na nim wywolywalibysmy metody zarzadzania danymi uzytkownika.

Wtedy metody: onUserJoinChatRoom, onUserDisconnect, onUserUnsubscribe do wywalenia i controller bezposrednio na chatSessionTracker by sobie wywolywal metody.
Problem widze w onUserUnsubscribe  bo tam usuwamy dane uzytkownika plus wysylasz wiadomosc websocketem. Ale moznaby to bylo zrobic tez w controllerze, czyli usunac dane poprzez bezposrednio przez chatSessionTracker i wyslac wiadomosc bezposrednio przez ChatSessionService. 
Mozesz dac znac co myslisz

Reply:
albo zostawic jak jest, ale wyodrebnic samo wywylanie wiadomosci osobno i tą klase z wysylaniem wstrzyknac dodatkowo do ChatSessionService.

Reply:
> Wtedy metody: onUserJoinChatRoom, onUserDisconnect, onUserUnsubscribe do wywalenia i controller bezposrednio na chatSessionTracker by sobie wywolywal metody.
Problem widze w onUserUnsubscribe bo tam usuwamy dane uzytkownika plus wysylasz wiadomosc websocketem. Ale moznaby to bylo zrobic tez w controllerze, czyli usunac dane poprzez bezposrednio przez chatSessionTracker i wyslac wiadomosc bezposrednio przez ChatSessionService.
Mozesz dac znac co myslisz

Czyli myślisz żeby te `onUserJoinChatRoom` itp. po prostu do kontrolera wrzucić tak?

Reply:
> albo zostawic jak jest, ale wyodrebnic samo wywylanie wiadomosci osobno i tą klase z wysylaniem wstrzyknac dodatkowo do ChatSessionService.

W zasadzie wszysko co robi ten `ChatSessionService` to jest wokół wysyłania. Ew. mógłbym metody w przeszukiwaniu po subskrypcjach przenieść do `WebSocketSessionService`, może wtedy by się trochę wyczyściło

Reply:
W sensie wywoływać bezpośrednio na tym chatSessionTrqcker W kontrolerze. Wtedy jest większa generycznosc bo chatSessionTracker będzie mógł być użyty poza chatSessionService gdyby była potrzeba, a tak to go wiążemy z chatSessionService

Reply:
w sumie można

Reply:
> Właśnie obecnie chatSessionService jeszcze obsługuje dodawanie subskrybcji czatu do mapy w chatSessionTracker i chodzi o to, żeby tego nie robił u zajął się tylko wysyłaniem wiadomości, a nie zarządzaniem stanem mapy w chatSessionTracker

W sumie będzie to sprytniejsze